### ISSUE TITLE
fatal error: Double value cannot be converted to Int because it is either infinite or NaN

### ISSUE THREAD

**Issue by dhamukrish** [danielgindi/Charts#1550]
When I use LineChart APIs, I am getting fatal error at ChartUtils.swift @ line 53.

`fatal error: Double value cannot be converted to Int because it is either infinite or NaN`

```
    internal class func decimals(_ number: Double) -> Int
    {
        if number.isNaN || number.isInfinite || number == 0.0
        {
            return 0
        }
        
        let i = roundToNextSignificant(number: Double(number))
        **return Int(ceil(-log10(i))) + 2**
    }
```

**Comment by dhamukrish** [comment]
I am now checking array elements count before passing to Charts API as workaround.

**Comment by petester42** [comment]
What double value are you passing? I'll write some test cases to try to fix this.

**Comment by dhamukrish** [comment]
@petester42 I was passing array of Double but array is empty.

**Comment by petester42** [comment]
so `let x:[Double] = []` or an array of empty values?

**Comment by dhamukrish** [comment]
`let x: [Double] = []` is passed without appending.

**Comment by petester42** [comment]
Then my assumption would be that the function would never be called. Can you provide the snippit you use to create this chart so I can reproduce it in my environment?

**Comment by dhamukrish** [comment]
**Previous: this produces fatal error when array is empty**
`setChart(lineChartSpeed, dataPoints: counterDrive, values: speedDrive)`

**Current one with condition to avoid this fatal error**
```
if counterDrive.count > 0 {
     setChart(lineChartSpeed, dataPoints: counterDrive, values: speedDrive)
}
```
**Function to set the graph datasets**
```
fileprivate func setChart(_ lineChartView: LineChartView, dataPoints: [Double], values: [Double]) {
    
    var dataEntries: [ChartDataEntry] = []
    
    for i in 0..<dataPoints.count {
      let dataEntry = ChartDataEntry(x: Double(i), y: values[i])
      dataEntries.append(dataEntry)
    }
    
    let lineChartDataSet = LineChartDataSet(values: dataEntries, label: "Speed")
    lineChartDataSet.setColor(UIColor.blue)
//    lineChartDataSet.drawCubicEnabled = true
    lineChartDataSet.mode = .cubicBezier
    lineChartDataSet.drawCirclesEnabled = false
    lineChartDataSet.lineWidth = 1.0
    lineChartDataSet.circleRadius = 5.0
    lineChartDataSet.highlightColor = UIColor.red
    lineChartDataSet.drawHorizontalHighlightIndicatorEnabled = true
    
    var dataSets = [IChartDataSet]()
    dataSets.append(lineChartDataSet)
    
    let lineChartData = LineChartData(dataSets: dataSets)
    
    lineChartView.data = lineChartData
  }

```

**Comment by petester42** [comment]
Yup confirmed bug. Should be a simple fix.

**Comment by dhamukrish** [comment]
@petester42 Thanks for this update and will wait for the fix.

**Comment by petester42** [comment]
fixed by #1558

**Comment by josman185hotm** [comment]
I have found the solution to this. Just be careful with the implementation of this pod, specifically in this method: 
func setChart(dataPoints: [String], values: [Double]) {
        ...
        ...
         let chartDataSet = LineChartDataSet(yVals: dataEntries, label: "")
        data = LineChartData(xVals: dataPoints, dataSet: chartDataSet)
}

Here the new implementation:
Make sure you are passing the "chartDataSet" instead of the "dataPoints".

func setChart(_ dataPoints: [String], values: [Double]) {
        ...
        ...
         let chartDataSet = LineChartDataSet(values: dataEntries, label: "")
        data = LineChartData(dataSet: chartDataSet)
}

I was making a big mistake passing the "dataPoints" instead of the "chartDataSet" values. Obviously Xcode doesn´t show any error until you run your code. 
Hope this helps.
